docs/user_manual.md

```diff
@@ -1007,6 +1008,43 @@ value given in the `configuration ...` declaration.
 Rule Declaration
 ----------------
 
+### Rule Structure
+
+Each K rule follows the same basic structure (given as an example here):
+
+```k
+rule LHS => RHS requires REQ ensures ENS [ATTRS]
+```
+
+The portion between `rule` and `requires` is referred to as the *rule body*,
+and may contain one or more rewrites (though not nested). Here, the rule body is
+`LHS => RHS`, where `LHS` and `RHS` are used as placeholders for the pre- and
+post- states. Note that we lose no generality referring to _the_ `LHS` or _the_
+`RHS`, even in the presence of multiple rewrites, as the rewrites are pulled to
+the top-level anyway.
+
+Next is the *requires clause*, represented here as `REQ`. The requires clause is
+an additional predicate (function-like term of sort `Bool`), which is to be
+evaluated before applying the rule. For functional rules, if the requires clause
+does not evaluate to `true`, then the rule does not apply. For semantic rules,
+the requires clause may cause the state to branch into two states, one where
+`REQ` holds and one where it does not.
```

Comment:
It says directly there, "one where `REQ` holds and one where it does not". I could add something saying "this corresponds to exploring both possibilities". I can also just remove the mention of branching, since unification with teh LHS can cause branching too and I don't mention that.

Reply:
Some explanation of branching behavior could improve this.
As you pointed out, not sure if this is the best place to do it though.

Reply:
I simplified the explanation, just took out the part about branching.